File: ios/com/lightningkite/kwifttemplate/shared/views/ExampleContentViewGenerator.swift
//Package: com.lightningkite.kwifttemplate.shared.views
//Converted using Kwift2

import Foundation



public class ExampleContentViewGenerator: ViewGenerator {
    
    public var stack: ObservableStack<ViewGenerator>
    
    override public var title: String {
        get {
            return "Example Content"
        }
    }
    public var number: StandardObservableProperty<Int32>
    
    public func increment() -> Void {
        number.value += 1
    }
    
    public func goToAnotherScreen() -> Void {
        stack.push(SwitchManiaVG(stack))
    }
    
    override public func generate(dependency: ViewDependency) -> View {
        var xml = ExampleContentXml()
        var view = xml.setup(dependency)
        xml.exampleContentIncrement.onClick(captureWeak(self) { (self) in 
            self.increment()
        })
        xml.exampleContentGoToAnotherScreen.onClick(captureWeak(self) { (self) in 
            self.goToAnotherScreen()
        })
        xml.exampleContentNumber.bindText(number) { (it) in 
            it.toString()
        }
        return view
    }
    override public func generate(_ dependency: ViewDependency) -> View {
        return generate(dependency: dependency)
    }
    
    public init(stack: ObservableStack<ViewGenerator>) {
        self.stack = stack
        self.number = StandardObservableProperty(0)
        super.init()
    }
    convenience public init(_ stack: ObservableStack<ViewGenerator>) {
        self.init(stack: stack)
    }
}
 

File: android/app/src/main/java/com/lightningkite/kwifttemplate/shared/views/ExampleContentViewGenerator.kt
package com.lightningkite.kwifttemplate.shared.views

import android.view.View
import com.lightningkite.kwift.observables.actual.bindText
import com.lightningkite.kwift.observables.shared.ObservableStack
import com.lightningkite.kwift.observables.shared.StandardObservableProperty
import com.lightningkite.kwift.shared.captureWeak
import com.lightningkite.kwift.views.actual.ViewDependency
import com.lightningkite.kwift.views.actual.onClick
import com.lightningkite.kwift.views.shared.ViewGenerator
import com.lightningkite.kwifttemplate.xml.ExampleContentXml

class ExampleContentViewGenerator(val stack: ObservableStack<ViewGenerator>) : ViewGenerator() {
    override val title: String get() = "Example Content"

    val number: StandardObservableProperty<Int> = StandardObservableProperty(0)

    fun increment(){
        number.value += 1
    }

    fun goToAnotherScreen(){
        stack.push(SwitchManiaVG(stack))
    }

    override fun generate(dependency: ViewDependency): View {
        val xml = ExampleContentXml()
        val view = xml.setup(dependency)
        xml.exampleContentIncrement.onClick(captureWeak(this) { self -> self.increment() })
        xml.exampleContentGoToAnotherScreen.onClick(captureWeak(this) { self -> self.goToAnotherScreen() })
        xml.exampleContentNumber.bindText(number) { it -> it.toString() }
        return view
    }
}
